src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -138,62 +156,138 @@
  *
  * <h2 id="slicing">Slicing memory segments</h2>
  *
- * Memory segments support <em>slicing</em>. A memory segment can be used to {@linkplain MemorySegment#asSlice(long, long) obtain}
- * other segments backed by the same underlying memory region, but with <em>stricter</em> spatial bounds than the ones
- * of the original segment:
+ * Memory segments support {@linkplain MemorySegment#asSlice(long, long) slicing}. Slicing a memory segment
+ * returns a new memory segment that is backed by the same region of memory as the original. The address of the sliced
+ * segment is derived from the address of the original segment, by adding an offset (expressed in bytes). The size of
+ * the sliced segment is either derived implicitly (by subtracting the specified offset from the size of the original segment),
+ * or provided explicitly. In other words, a sliced segment has <em>stricter</em> spatial bounds than those of the original segment:
```

Comment:
Strictly, a sliced segment can have the *same* spatial bounds as the original segment.

Reply:
True - but I think the current text is a good compromise e.g. it is narrative text. You can dive into `MemorySegment::asSlice` and find out more.